python/ray/data/_internal/execution/autoscaler/autoscaler.py

```diff
@@ -0,0 +1,43 @@
+from abc import ABCMeta, abstractmethod
+from typing import TYPE_CHECKING
+
+from ray.util.annotations import DeveloperAPI
+
+if TYPE_CHECKING:
+    from ray.data._internal.execution.resource_manager import ResourceManager
+    from ray.data._internal.execution.streaming_executor_state import (
+        SchedulingDecision,
+        Topology,
+    )
+
+
+@DeveloperAPI
+class Autoscaler(metaclass=ABCMeta):
+    """Abstract interface for Ray Data autoscaler."""
+
+    def __init__(
+        self,
+        topology: "Topology",
+        resource_manager: "ResourceManager",
+        execution_id: str,
+    ):
+        self._topology = topology
+        self._resource_manager = resource_manager
+        self._execution_id = execution_id
+
+    @abstractmethod
+    def try_trigger_scaling(self, scheduling_decision: "SchedulingDecision"):
```

Comment:
This method is supposed to trigger autoscaling of both the cluster and actor pools, right? Could we add a note in the docstring? I don't think it's obvious what's getting autoscaled from the name